### PR TITLE
Add support for ncrystal materials

### PR DESCRIPTION
**Pull request by zoeprieto**
This pull request aims to fix the issue #2422. The changes are simple but one remaining point would be to obtain the maximum NCrystal energy from the C API. @paulromano how can we access the API to obtain NCRYSTAL_MAX_ENERGY?

### CODE REVIEW

**Comment by zoeprieto** [comment]
```
cfg = 'LiquidWaterH2O_T293.6K.ncmat'
m1 = openmc.Material.from_ncrystal(cfg)
m1.name = cfg
openmc.plot_xs(m1, ('scatter',))
```
image


Here there is a simple example

File: openmc/plotter.py [hunk 1]
```diff
@@ -438,6 +439,19 @@ def _calculate_cexs_nuclide(this, types, temperature=294., sab_name=None,
                             [sab_sum, nuc[mt].xs[nucT]],
                             [sab_Emax])
                         funcs.append(pw_funcs)
+                    elif ncrystal_cfg:
+                        import NCrystal
+                        nc_scatter = NCrystal.createScatter(ncrystal_cfg)
+                        nc_func = nc_scatter.crossSectionNonOriented
+                        nc_emax = 5 # eV # this should be obtained from NCRYSTAL_MAX_ENERGY
```
Comment (paulromano) [inline openmc/plotter.py:446]: I think there was some discussion about having the max energy be configurable, but I forget where we ended up on this. @tkittel Do you recall? In any event, I'm fine with the hard-coded value for now.

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for implementing this @zoeprieto! It looks like there is a conflict in plotter.py due to the recently merged PR #2413, so you'll need to merge in the latest develop branch into your branch and resolve the conflicts.

**Comment by zoeprieto** [comment]
Hey @paulromano , I incorporated the changes to my branch. Can you please review it again?

**Review by paulromano — APPROVED** [review]
Thanks @zoeprieto!

**Pull request merged**